python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -317,6 +335,24 @@ def read_messages(
     def on_opened(self, opened_payload: PipesOpenedData) -> None:
         self.opened_payload = opened_payload
 
+    def update_params(
```

Comment:
Generally prefer to avoid this sort of mutability if possible.

Reply:
Should we store these provided params in a separate attribute? 
The whole point is to be able to receive additional params provided manually.

Reply:
I'm working on an example of `update_params` usage with `PipesCloudWatchMessageReader` as a separate PR